dscache/dscache.go

```diff
@@ -173,6 +174,37 @@ func (d *Dscache) ListRefs() ([]reporef.DatasetRef, error) {
 	return refs, nil
 }
 
+// ResolveRef finds the identifier for a dataset reference
+// implements resolve.RefResolver interface
+func (d *Dscache) ResolveRef(ctx context.Context, ref *dsref.Ref) error {
+	// NOTE: isEmpty is nil-callable
+	if d.IsEmpty() {
+		return resolve.ErrCannotResolveName
+	}
+
+	// Handle the "me" convenience shortcut
+	if ref.Username == "me" && d.DefaultUsername != "" {
```

Comment:
I don't think this belongs in dscache. Handling "me" is a surface-level UI convenience, and it shouldn't be known about by lower-level components. While investigating the save path, I found tons of subtle bugs around how base handles blank and "me" usernames, including unit tests that rely on strange and broken behavior, as a consequence of "me" handling showing up throughout the stack. Better to keep the lower-level of the code simple, straight-forward, non-magical, and easy to reason about.

Reply:
totally. felt icky while putting it in, I'll move it up